docs/json.md

```diff
@@ -0,0 +1,421 @@
+
+Serialization is the conversion of a JSON payload into a structure, record, or object of a given language. Deserialization is rendering an instantiated data type to its JSON representation.
+
+Almost all languages include JSON serialization and deserialization support. Many include more than one, with an offering from a third-party as the de facto standard, rather than the first-party solution provided by the language vendor.
+
+The parsing requirements of concern for the Looker language SDKs are:
+
+1. JSON value types conflicting with the API type specification. This is called **type ambiguity** below. For the Looker API, these conflicts should be automatically resolvable for our SDKs. The conflict is either:
+   1. a numeric value for a string property (forward compatibility issue)
+   2. a string value for an integer property (backward compatibility issue)
+2. New properties in a JSON payload. These are ignored by the SDKs. (See [TypeScript caveats](#typescript-caveats) for additional comments.)
+3. Removed properties in a JSON payload. Typically, the property will end up unassigned. Some SDKs may throw errors with missing required properties, but that client code should be changed anyway if the property no longer exists.
+4. Name changes, like `space` to `folder` and `homepage` to `board`. There is no reasonable automatic solution for this, so type or property renames will require changing SDK existing code.
+
+### forward and backward compatibility
+
+In this document, **forward compatible** means an older SDK can process payloads from a newer Looker API server.
+
+**Backward compatible** means a newer SDK can process payloads from an older Looker API server.
+
+Our goal with the Looker language SDKs is to provide both forward and backward compatibility.
+
+The current Looker API server is written in Ruby, which is one of the aforementioned "flexibly typed" languages. The Looker API server accepts ID value references as **either** `integer` or `string` values and converts them internally to the required type.
+This means the LookerAPI server is fully compatible with any Looker SDK for identically named properties and parameters that are integer in an older version and string in a newer version.
+
+### type ambiguity
+
+With API 3.1, an internal numeric ID value is often returned with `integer` value syntax even though the API specification described the property type as `string`. For API 4.0, the API server always returns property types that comply with the API type specification.
+
+For full backward and forward compatibility between the SDK and the Looker API, the SDK should accept either a `string` or `integer` JSON value and convert it to the required target type of either `integer` or `string`.
+
+We've added unit tests for the language SDKs to verify this behavior. This is a sample JSON payload used in many of the tests:
```

Comment:
```suggestion
Looker SDKs have unit tests to verify this behavior. This is a sample JSON payload used in the tests:
```

Reply:
sample payload we used in many of the tests, not all